### PR TITLE
Wpp 131 google bug

### PR DESCRIPTION
**Pull request by Maxwell1455**
## For Pull Requesters/Reviewers:
All of the following should be true before making a pull request and should be inspected by reviewers:

### User Story
- [ ] Acceptance criteria are adhered to and completely finished
- [ ] All subtasks on Jira are completed

### Unit Testing
- [ ] Unit tests are finished and passing
- [ ] Code coverage is >80%

### Documentation/Style
- [ ] JSDocs are written for methods and components
- [ ] Unclear code has been documented
- [ ] Style standards are adhered to

## For Reviewers:
The following should be additionally checked by reviewers:

- [ ] Code runs locally
- [ ] Visually inspect tests; they should not simply pass, but rather they should be doing something and testing acceptance criteria
- [ ] See if "jank" code could be rewritten

## Description
Insert here the description of what your PR does.

### CODE REVIEW

File: server/db.js [hunk 1]
```diff
@@ -25,7 +25,8 @@ function initDatabase(fp, puzzles) {
         user_id INTEGER PRIMARY KEY AUTOINCREMENT,
         username TEXT UNIQUE,
         hashed_password BLOB,
-        salt BLOB
+        salt BLOB, \
+        is_oauth INTEGER \
```
Comment (Blackkirby72) [inline server/db.js]: could honestly take the `is_oath` column out of here
also, the query is now a formatted string, so whitespace is preserved. no \ needed

File: server/api/googleLogin.js [hunk 2]
```diff
@@ -71,7 +71,7 @@ async function googleSignup(req, res, username) {
  */
 async function insertGoogleUser(db, username, callback) {
   await db.run(
-    "INSERT INTO User (username) VALUES (?)",
+    "INSERT INTO User (username, is_oauth) VALUES (?, 1)",
```
Comment (Blackkirby72) [inline server/api/googleLogin.js]: I feel like we're going to want a separate column to store the Google ID. 

WPP-137 will allow users to connect their Google Account to a username/password combo, so we need a way to store both the username and the Google ID. I'd just leave the `username` as null 

If we leave it as is, the user signs in with Google for the first time => creates a username/password combo => that username overwrites the Google ID => the user can no longer sign in with Google

File: server/db.js [hunk 3]
```diff
@@ -81,6 +82,7 @@ function initDatabase(fp, puzzles) {
 const addColumns = (db) => {
   const columnsToAdd = {
     User: [
+      { name: "is_oauth", typeDef: "INTEGER"},
```
Comment (Blackkirby72) [inline server/db.js]: Probably rename to something like `oauth_id` or the like. Then store the Google ID in this field

**Review by Blackkirby72 — CHANGES_REQUESTED** [review]

File: server/db.js (hunk 1 above)
Comment (Maxwell1455) [reply server/db.js]: Done.

File: server/api/googleLogin.js (hunk 2 above)
Comment (Maxwell1455) [reply server/api/googleLogin.js]: Done x2.

File: server/db.js (hunk 3 above)
Comment (Maxwell1455) [reply server/db.js]: Done x3.

**Review by prewe5294 — APPROVED** [review]
works locally now, the column was made, so looks good to me

File: server/api/googleLogin.js [hunk 4]
```diff
@@ -18,7 +18,7 @@ async function googleLogin(req, res, next) {
   const username = req.body.googleIdToken;
```
Comment (Blackkirby72) [inline server/api/googleLogin.js]: `username` -> `oauth_id` bit of a semantics thing, but looks good otherwise

Comment (Blackkirby72) [reply server/api/googleLogin.js]: make sure to update it everywhere else in this file too if you update it. I would send back a `null` username in the response for now, and we probably don't need the `oauth_id` on the front end

**Review by Blackkirby72 — APPROVED** [review]

**Pull request merged**